File: method_chaining.cpp
#include <iostream>

class Calculator {
private:
    double result;

public:
    Calculator() : result(0) {}

    Calculator& add(double value) {
        result += value;
        return *this;
    }

    Calculator& subtract(double value) {
        result -= value;
        return *this;
    }

    Calculator& multiply(double value) {
        result *= value;
        return *this;
    }

    Calculator& divide(double value) {
        if (value != 0) result /= value;
        return *this;
    }

    double get_result() const {
        return result;
    }
};

int main() {
    Calculator calc;
    double result = calc.add(5)
                        .multiply(2)
                        .subtract(3)
                        .divide(2)
                        .get_result();
    std::cout << result << std::endl;  // 출력: 3.5
}

File: csharp/method_chaining.cs
using System;

class Calculator
{
    private double result = 0;

    public Calculator Add(double value)
    {
        result += value;
        return this;
    }

    public Calculator Subtract(double value)
    {
        result -= value;
        return this;
    }

    public Calculator Multiply(double value)
    {
        result *= value;
        return this;
    }

    public Calculator Divide(double value)
    {
        if (value != 0)
            result /= value;
        return this;
    }

    public double GetResult()
    {
        return result;
    }
}

class Program
{
    static void Main()
    {
        var calc = new Calculator();
        double result = calc.Add(5)
                            .Multiply(2)
                            .Subtract(3)
                            .Divide(2)
                            .GetResult();
        Console.WriteLine(result); // 출력: 3.5
    }
}
// 